src/logic/basic.lean

```diff
@@ -767,6 +767,18 @@ by simp [@eq_comm _ a']
   (∀ b, ∀ a, f a = b → p b) ↔ (∀ a, p (f a)) :=
 by { rw forall_swap, simp }
 
+@[simp] theorem forall_eq_apply_imp_iff {f : α → β} {p : β → Prop} :
+  (∀ a, ∀ b, b = f a → p b) ↔ (∀ a, p (f a)) :=
+by simp [@eq_comm _ _ (f _)]
+
+@[simp] theorem forall_eq_apply_imp_iff' {f : α → β} {p : β → Prop} :
+  (∀ b, ∀ a, b = f a → p b) ↔ (∀ a, p (f a)) :=
+by { rw forall_swap, simp }
+
+/-- An implication whose LHS is independent of the forall can be brought outside -/
+@[simp] theorem forall_imp_indep {p : Prop} {q : α → Prop} :
+  (∀ a, p → q a) ↔ (p → ∀ a, q a) := forall_swap
```

Comment:
This one feels risky, but I thought I'd try it

Reply:
Failed at https://github.com/leanprover-community/mathlib/blob/3419986967dad05d9f89b9c2b13011ee4313575c/src/data/set/basic.lean#L455 with a timeout